solc/CommandLineInterface.cpp

```diff
@@ -728,7 +754,14 @@ bool CommandLineInterface::compile()
 			m_compiler->setParserErrorRecovery(m_options.input.errorRecovery);
 		}
 
-		bool successful = m_compiler->compile(m_options.output.stopAfter);
+		bool successful;
+		if (m_options.output.stopAfter.has_value())
+			successful = m_compiler->compile(m_options.output.stopAfter.value());
+		else
+		{
+			m_options.output.stopAfter = CompilerStack::State::CompilationSuccessful;
+			successful = m_compiler->compile(m_options.output.stopAfter.value());
+		}
```

Comment:
`CommandLineInterface` should not modify `m_options`. Please initialize it in `CommandLineParser` (but only in compiler mode; in other modes an attempt to initialize it should cause an error but that will be covered in #11629).